src/vos/tests/vts_dtx.c

```diff
@@ -168,7 +168,7 @@ vts_dtx_begin(struct dtx_id *xid, daos_unit_oid_t *oid, daos_handle_t coh,
 	      daos_epoch_t epoch, uint64_t dkey_hash,
 	      struct dtx_conflict_entry *conflict, struct dtx_id *dti_cos,
 	      int dti_cos_count, uint32_t pm_ver, uint32_t intent,
-	      struct dtx_handle **dthp)
+	      bool is_leader, struct dtx_handle **dthp)
```

Comment:
The @is_leader is unnecessary, in the dtx vos test, we can always set current replica as leader. For the non-leader case, we can use the fail_loc (DAOS_VOS_NON_LEADER) as you added in the patch.